### PR TITLE
catch oauth error and set as generator.res

### PR DESCRIPTION
**Pull request by iamigo**
so that handleCollectResponse will generate error samples

The crux of the change is catching the error thrown by oauth2[method].getToken(simpleOauth.tokenConfig). The rest of the changes are minor cleanup.

### CODE REVIEW

File: test/remoteCollection/collect.js [hunk 1]
```diff
@@ -574,7 +573,7 @@ describe('test/remoteCollection/collect.js >', () => {
         expect(collectRes.res).to.not.equal(undefined);
         expect(collectRes.res.status).to
           .equal(httpStatus.SERVICE_UNAVAILABLE);
-        expect(collectRes.res.body.error.message)
```
Comment (jgraff2) [inline test/remoteCollection/collect.js]: Why this change?

File: src/remoteCollection/collect.js [hunk 2]
```diff
@@ -151,79 +152,90 @@ function prepareRemoteRequest(generator) {
...
+        // eslint-disable-next-line global-require
+        const oauth2 = require('simple-oauth2').create(simpleOauth.credentials);
+        return oauth2[method].getToken(simpleOauth.tokenConfig)
+          .catch((err) => {
+            // Setting this Oauth error as the generator "res" attribute to
+            // force handleCollectResponse to generate error samples with the
+            // error message.
+            debug('simple-oauth2 (method=%s)', method, err);
+            generator.res = err;
+          });
```
Comment (jgraff2) [inline src/remoteCollection/collect.js]: Won't the request still be sent?

**Comment by jgraff2** [comment]
Tests?
There are some existing tests in the integration testing that cover oauth. You should make sure those still pass with this change, and see if it makes sense to add one to cover this case.

File: test/remoteCollection/collect.js [hunk 3]
```diff
@@ -574,7 +573,7 @@ describe('test/remoteCollection/collect.js >', () => {
...
+        expect(collectRes.res.response.body.error.message)
```
Comment (jgraff2) [inline test/remoteCollection/collect.js]: Why this change?

File: src/remoteCollection/collect.js (hunk 2 above)
Comment (iamigo) [reply src/remoteCollection/collect.js]: yeah - I'll update to skip that

File: test/remoteCollection/collect.js (hunk 3 above)
Comment (iamigo) [reply test/remoteCollection/collect.js]: I had to change this test when I added the "else if" on line 139 of collect.js

**Comment by iamigo** [comment]
@jgraff2  re "some existing tests in the integration testing that cover oauth" -- see https://github.com/salesforce/refocus/pull/1111

File: src/remoteCollection/collect.js [hunk 4]
```diff
@@ -135,7 +135,13 @@ function sendRemoteRequest(generator) {
         } // shouldRequestNewToken
 
         debug('sendRemoteRequest returned error %O', err);
-        generator.res = err;
+        if (get(generator, 'connection.simple_oauth')) {
+          const method = generator.connection.simple_oauth.method;
+          generator.res =
+            new Error(`simple-oauth2 (method=${method}): ${err.message}`);
```
Comment (jgraff2) [inline src/remoteCollection/collect.js]: Why did you add this here as well? Should it be different from the one below?

Comment (iamigo) [reply src/remoteCollection/collect.js]: This error is still an authentication error, based on the simple-oauth2 configuration, but the error here was an authentication error on the remote data source, while the other one was an error actually calling simple oauth's getToken function

**Comment by pallavi2209** [comment]
@iamigo Please merge.

**Comment by iamigo** [comment]
@pallavi2209 merged and published new version to npm
